rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -315,7 +317,8 @@ def decode_transaction(
             if event:
                 events.append(event)
 
-        self.run_all_post_decoding_rules(
+        events.sort(key=lambda x: x.sequence_index, reverse=False)
+        events = self.run_all_post_decoding_rules(
```

Comment:
why sort before and not after? Is it guaranteed that `run_all_post_decoding_rules` won't change order of events? or is it documented somewhere that it shouldn't?

Reply:
The reason why I made this change is because during the main decoding loop `maybe_reshuffle_events` may replace sequence indexes of some of events. But it doesn't re-order the events in the `decoded_events` list.

This change is not mandatory for uniswap routers post-decoders, but I just thought that it will be good if post-decoders receive events in the right order.

Should the events be re-sorted after running post-decoding rules or do you maybe think that this change is not needed at all?

Reply:
What you did for example can introduce an error by the developer on how the events get genearted and return them in the wrong order. I would sort at the end. About sorting before... maybe you can avoid it since is not required now?

Reply:
Perhaps yes